Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/SMAA.azsli

```diff
@@ -1135,13 +1135,11 @@ float4 SMAABlendingWeightCalculationPS(float2 texcoord,
         if (!o_enableDiagonalDetectionFeature ||
             weights.r == -weights.g) // weights.r + weights.g == 0.0
         {
-            float2 d;
-
             // Find the distance to the left:
             float3 coords;
             coords.x = SMAASearchXLeft(SMAATexturePass2D(edgesTex), SMAATexturePass2D(searchTex), offset[0].xy, offset[2].x);
             coords.y = offset[1].y; // offset[1].y = texcoord.y - 0.25 * SMAA_RT_METRICS.y (@CROSSING_OFFSET)
-            d.x = coords.x;
+            float dx = coords.x;
```

Comment:
Add a small comment saying
//Do not combine coords.x and coords.z as float2 causes problems on Mali devices.

Reply:
Added comments at the top of the scope